app/models/services/locators/doi_paper_locator.rb

```diff
@@ -6,30 +6,32 @@ def column
   def find_paper
     return super if super
 
+    paper = Paper.new
+
+    crossref = Crossref.new
+    if result = crossref.get_doi_metadata(self.locator_id)
+      data = result['message']
+      paper = self.import_data_to_paper(paper,data,'crossref')
+    end
+
     pubmed = Pubmed.new
     if uid = pubmed.find_uid_by_doi(self.locator_id)
       result = pubmed.get_uid_metadata(uid)
       data = result['result'][uid]
-      names = data['authors'].map {|a| a['name']}
+      paper = self.import_data_to_paper(paper,data,'pubmed')
+    end
 
-      existing_authors = Author.where(name: names)
-      existing_names = existing_authors.map(&:name)
-      new_authors = (names - existing_names).map {|a| Author.create name: a}
+    existing_paper = Paper.where( "title = ? or pubmed_id = ? and pubmed_id is not null", paper.title, paper.pubmed_id )
 
-      paper = Paper.create(
-        pubmed_id: data['uid'],
-        title: data['title'],
-        published_at: data['pubdate'],
-        authors: (existing_authors + new_authors),
-        abstract: pubmed.get_abstract(data['uid']),
-        doi: self.locator_id,
-        publication: data['source']
-      )
+    if existing_paper.present?
+      return existing_paper
```

Comment:
yea, I chose where because I couldn't find an easy way to do an "or" statement for 2 attributes using #find.  It was easier to do the "or" in the where statement to broaden and eliminate duplication.  

The "or" is for the case where they may have entered a paper title without a pubmed id or doi, or if the title was imported slightly differently. I'll test your suggestion to see if I get the same result.